Vic2ToHoI4/Source/OutHoi4/OutHoi4World.cpp

```diff
@@ -0,0 +1,549 @@
+	outputIdeologies(world.getIdeologies(), outputName);
+	outputLeaderTraits(world.getIdeologicalLeaderTraits(), world.getMajorIdeologies(), outputName);
+	outIdeas(world.getTheIdeas(), world.getMajorIdeologies(), outputName);
+	outputBookmarks(world.getGreatPowers(), world.getCountries(), outputName);
+	outputScriptedLocalisations(outputName, world.getScriptedLocalisations());
+	outputScriptedTriggers(world.getScriptedTriggers(), outputName);
+	outputDifficultySettings(world.getGreatPowers(), outputName);
+	outputGameRules(world.getGameRules(), outputName);
+}
+
+
+void HoI4::outputCommonCountries(const std::map<std::string, std::shared_ptr<Country>>& countries,
+	 const std::string& outputName)
+{
+	LOG(LogLevel::Info) << "\t\tCreating country tags";
+	if (!Utils::TryCreateFolder("output/" + outputName + "/common/country_tags"))
+	{
+		throw std::runtime_error("Could not create output/" + outputName + "/common/country_tags");
+	}
+
+	std::ofstream allCountriesFile("output/" + outputName + "/common/country_tags/00_countries.txt");
+	if (!allCountriesFile.is_open())
+	{
+		throw std::runtime_error("Could not create output/" + outputName + "/common/country_tags/00_countries.txt");
+	}
+
+	for (const auto& country: countries)
+	{
+		if (country.second->isGreatPower() && country.second->getCapitalState())
+		{
+			outputToCommonCountriesFile(allCountriesFile, *country.second);
+		}
+	}
+
+	for (const auto& country: countries)
+	{
+		if (!country.second->isGreatPower() && country.second->getCapitalState())
+		{
+			outputToCommonCountriesFile(allCountriesFile, *country.second);
+		}
+	}
+
+	allCountriesFile << "\n";
+	allCountriesFile.close();
+}
+
+
+void HoI4::outputColorsFile(const std::map<std::string, std::shared_ptr<Country>>& countries,
+	 const std::string& outputName)
+{
+	LOG(LogLevel::Info) << "\t\tWriting country colors";
+
+	if (!Utils::TryCreateFolder("output/" + outputName + "/common/countries"))
+	{
+		throw std::runtime_error("Could not create output/" + outputName + "/common/countries");
+	}
+
+	std::ofstream output("output/" + outputName + "/common/countries/colors.txt");
+	if (!output.is_open())
+	{
+		throw std::runtime_error("Could not open output/" + outputName + "/common/countries/colors.txt");
+	}
+
+	output << "#reload countrycolors\n";
+	for (const auto& country: countries)
+	{
+		if (country.second->getCapitalState())
+		{
+			outputColors(output, *country.second);
+		}
+	}
+
+	output.close();
+}
+
+
+void HoI4::outputNames(const namesMapper& theNames,
+	 const std::map<std::string, std::shared_ptr<Country>>& countries,
+	 const std::string& outputName)
+{
+	LOG(LogLevel::Info) << "\t\tWriting names";
+
+	std::ofstream namesFile("output/" + outputName + "/common/names/01_names.txt");
+	if (!namesFile.is_open())
+	{
+		throw std::runtime_error("Could not open output/" + outputName + "/common/names/01_names.txt");
+	}
+
+	namesFile << "\xEF\xBB\xBF"; // add the BOM to make HoI4 happy
```

Comment:
For heaven's sake there has to be a better method for this. Globally I mean, hoi4, imperator etc included.

Reply:
See response below.